src/Nethermind/Nethermind.Evm/TransactionSubstate.cs

```diff
@@ -58,6 +58,12 @@ public readonly ref struct TransactionSubstate
     public IToArrayCollection<LogEntry> Logs => _logs ?? _emptyLogs;
     public IHashSetEnumerableCollection<Address> DestroyList => _destroyList ?? _emptyDestroyList;
 
+    /// <summary>
+    /// Policy-specific data for this transaction (e.g., multigas for Arbitrum).
+    /// Null for standard Ethereum transactions using SimpleGasPolicy.
+    /// </summary>
+    public object? PolicyData { get; set; }
```

Comment:
Does this needs to be settable? Can't it come only on init in constructor?

We want to keep this struct `readonly` for performance reasons.